### PR TITLE
sti-assessments: add integration for views, fix more rspec tests

### PR DESCRIPTION
**Pull request by robert-tan**
Manual testing on web application, fixed most glaring issues that prevent routes from rendering

### CODE REVIEW

File: app/controllers/sections_controller.rb [hunk 1]
```diff
@@ -8,7 +8,7 @@ class SectionsController < ApplicationController
   # Displays sections, and allows to create them
   #TODO Displays metrics concerning users and sections
   def index
-    @sections = Section.all
+    @sections = Section.all.includes(:students)
```
Comment (david-yz-liu) [inline app/controllers/sections_controller.rb]: You should revert this change (because it's in a separate pull request).

File: app/views/assignments/_form.html.erb [hunk 2]
```diff
@@ -13,437 +13,438 @@
...
+                         onchange: 'set_onbeforeunload(true);',
+                         value: @assignment.due_date.nil? ? '' :
+                                @assignment.due_date.strftime(
+                                  '%Y-%m-%d %l:%M %p'),
+                         size: 35,
+                         required: true,
+                         placeholder: t('time.format_string.placeholder') %>
+      </div>
+
+      <% if !@assignment.assignment_properties.scanned_exam %>
```
Comment (david-yz-liu) [inline app/views/assignments/_form.html.erb]: Use `unless` here.

File: app/views/assignments/_form.html.erb [hunk 3]
```diff
@@ -13,437 +13,438 @@
...
+      <div>
+        <%= ap_f.check_box :allow_web_submits,
+                        onclick: 'set_onbeforeunload(true);' %>
```
Comment (david-yz-liu) [inline app/views/assignments/_form.html.erb]: Align this line with the `:allow_web_submits` above.

File: app/views/assignments/_form.html.erb [hunk 4]
```diff
@@ -13,437 +13,438 @@
...
+        <%= ap_f.label :allow_web_submits,
+                    Assignment.human_attribute_name(:allow_web_submits) %>
```
Comment (david-yz-liu) [inline app/views/assignments/_form.html.erb]: Similar thing here.

Comment (david-yz-liu) [reply app/views/assignments/_form.html.erb]: (And throughout this file.)

File: app/views/grade_entry_forms/_form.html.erb [hunk 5]
```diff
@@ -45,7 +45,7 @@
 
       <div>
         <%= f.label :date, GradeEntryForm.human_attribute_name(:date) %>
```
Comment (david-yz-liu) [inline app/views/grade_entry_forms/_form.html.erb]: This also should change to `:due_date`

File: app/controllers/groups_controller.rb [hunk 6]
```diff
@@ -172,9 +172,9 @@ def assign_scans
   def get_names
     names = Student
               .select(:id, :id_number, :user_name, "CONCAT(first_name,' ',last_name) AS label, CONCAT(first_name,' ',last_name) AS value")
-              .where("(lower(first_name) like ? OR lower(last_name) like ? OR lower(user_name) like ? OR id_number like ?) AND users.id NOT IN (?)",
+              .where('(lower(first_name) like ? OR lower(last_name) like ? OR lower(user_name) like ? OR id_number like ?) AND users.id NOT IN (?)',
                      "#{params[:term].downcase}%", "#{params[:term].downcase}%", "#{params[:term].downcase}%", "#{params[:term]}%",
-                     Membership.select(:user_id).joins(:grouping).where("groupings.assignment_id = ?", params[:assignment]));
+                     Membership.select(:user_id).joins(:grouping).where('groupings.assessment_id = ?', params[:assignment]))
```
[the remark this answers is not in the document]
Comment (robert-tan) [reply app/controllers/groups_controller.rb]: Ignoring

File: app/controllers/groups_controller.rb [hunk 7]
```diff
@@ -171,10 +171,10 @@ def assign_scans
...
   def get_names
     names = Student
-              .select(:id, :id_number, :user_name, "CONCAT(first_name,' ',last_name) AS label, CONCAT(first_name,' ',last_name) AS value")
-              .where("(lower(first_name) like ? OR lower(last_name) like ? OR lower(user_name) like ? OR id_number like ?) AND users.id NOT IN (?)",
-                     "#{params[:term].downcase}%", "#{params[:term].downcase}%", "#{params[:term].downcase}%", "#{params[:term]}%",
-                     Membership.select(:user_id).joins(:grouping).where("groupings.assignment_id = ?", params[:assignment]));
+            .select(:id, :id_number, :user_name, "CONCAT(first_name,' ',last_name) AS label, CONCAT(first_name,' ',last_name) AS value")
+            .where('(lower(first_name) like ? OR lower(last_name) like ? OR lower(user_name) like ? OR id_number like ?) AND users.id NOT IN (?)',
+                   "#{params[:term].downcase}%", "#{params[:term].downcase}%", "#{params[:term].downcase}%", "#{params[:term]}%",
+                   Membership.select(:user_id).joins(:grouping).where('groupings.assessment_id = ?', params[:assignment]))
```
[the remark this answers is not in the document]
Comment (robert-tan) [reply app/controllers/groups_controller.rb]: Ignoring

File: app/controllers/groups_controller.rb [hunk 8]
```diff
@@ -171,10 +171,10 @@ def assign_scans
 
   def get_names
     names = Student
-              .select(:id, :id_number, :user_name, "CONCAT(first_name,' ',last_name) AS label, CONCAT(first_name,' ',last_name) AS value")
-              .where("(lower(first_name) like ? OR lower(last_name) like ? OR lower(user_name) like ? OR id_number like ?) AND users.id NOT IN (?)",
-                     "#{params[:term].downcase}%", "#{params[:term].downcase}%", "#{params[:term].downcase}%", "#{params[:term]}%",
-                     Membership.select(:user_id).joins(:grouping).where("groupings.assignment_id = ?", params[:assignment]));
+            .select(:id, :id_number, :user_name, "CONCAT(first_name,' ',last_name) AS label, CONCAT(first_name,' ',last_name) AS value")
+            .where('(lower(first_name) like ? OR lower(last_name) like ? OR lower(user_name) like ? OR id_number like ?) AND users.id NOT IN (?)',
```
[the remark this answers is not in the document]
Comment (robert-tan) [reply app/controllers/groups_controller.rb]: Ignoring

File: app/controllers/groups_controller.rb [hunk 9]
```diff
@@ -171,10 +171,10 @@ def assign_scans
 
   def get_names
     names = Student
-              .select(:id, :id_number, :user_name, "CONCAT(first_name,' ',last_name) AS label, CONCAT(first_name,' ',last_name) AS value")
-              .where("(lower(first_name) like ? OR lower(last_name) like ? OR lower(user_name) like ? OR id_number like ?) AND users.id NOT IN (?)",
-                     "#{params[:term].downcase}%", "#{params[:term].downcase}%", "#{params[:term].downcase}%", "#{params[:term]}%",
-                     Membership.select(:user_id).joins(:grouping).where("groupings.assignment_id = ?", params[:assignment]));
+            .select(:id, :id_number, :user_name, "CONCAT(first_name,' ',last_name) AS label, CONCAT(first_name,' ',last_name) AS value")
```
[the remark this answers is not in the document]
Comment (robert-tan) [reply app/controllers/groups_controller.rb]: Ignoring

File: app/controllers/groups_controller.rb [hunk 10]
```diff
@@ -171,10 +171,10 @@ def assign_scans
...
+            .where('(lower(first_name) like ? OR lower(last_name) like ? OR lower(user_name) like ? OR id_number like ?) AND users.id NOT IN (?)',
+                   "#{params[:term].downcase}%", "#{params[:term].downcase}%", "#{params[:term].downcase}%", "#{params[:term]}%",
```
[the remark this answers is not in the document]
Comment (robert-tan) [reply app/controllers/groups_controller.rb]: Ignoring
